Jenkinsfile

```diff
@@ -1,46 +1,83 @@
 #!/usr/bin/env groovy
 
 node('rhel7'){
-	stage('Checkout repo') {
+	stage('Build RSP Server') {
 		deleteDir()
-		git url: 'https://github.com/redhat-developer/vscode-adapters.git'
+		dir('rsp-server') {
+			git url: 'https://github.com/redhat-developer/rsp-server.git'
+
+			def mvnHome = tool 'maven3-latest'
+			env.PATH="${env.PATH}:${mvnHome}/bin"
+
+			sh 'mvn clean install'
+			sh 'mvn clean package -f distribution/pom.xml'
+		}
 	}
 
-	stage('Install requirements') {
-		def nodeHome = tool 'nodejs-8.11.1'
-		env.PATH="${env.PATH}:${nodeHome}/bin"
-		sh "npm install -g typescript vsce"
+	stage('Setup') {
+		dir('adapters') {
+			git url: 'https://github.com/redhat-developer/vscode-adapters.git'
+		
+			def nodeHome = tool 'nodejs-8.11.1'
+			env.PATH="${env.PATH}:${nodeHome}/bin"
+			sh "npm install -g typescript vsce"
+		}
 	}
 
-	stage('Build') {
-		sh "npm install"
-		sh "npm run package"
-		sh "npm run vscode:prepublish"
+	stage('Build Extension') {
+		dir('adapters') {
+			sh "npm install"
+			sh "unzip ${WORKSPACE}/rsp-server/distribution/target/org.jboss.tools.rsp.distribution*.zip"
+			sh "mv rsp-distribution server"
+			sh "npm run vscode:prepublish"
+		}
 	}
 
-	withEnv(['JUNIT_REPORT_PATH=report.xml', 'CODE_TESTS_WORKSPACE=c:/unknown']) {
-        stage('Test') {
-    		wrap([$class: 'Xvnc']) {
-    			sh "npm test --silent"
-    			//cobertura coberturaReportFile: 'coverage/cobertura-coverage.xml'
-    			junit 'report.xml'
-    		}
+	withEnv(['JUNIT_REPORT_PATH=report.xml']) {
+        stage('Test Extension') {
+			dir('adapters') {
+				wrap([$class: 'Xvnc']) {
+					sh "npm test --silent"
+					//cobertura coberturaReportFile: 'coverage/cobertura-coverage.xml'
+					junit 'report.xml'
+				}
+			}
         }
 	}
 
-	stage('Package') {
-		try {
+	stage('Package Extension') {
+		dir('adapters') {
 			def packageJson = readJSON file: 'package.json'
 			sh "vsce package -o adapters-${packageJson.version}-${env.BUILD_NUMBER}.vsix"
 		}
-		finally {
-			archiveArtifacts artifacts: '*.vsix'
-		}
 	}
+
 	if(params.UPLOAD_LOCATION) {
 		stage('Snapshot') {
-			def filesToPush = findFiles(glob: '**.vsix')
-			sh "rsync -Pzrlt --rsh=ssh --protocol=28 ${filesToPush[0].path} ${UPLOAD_LOCATION}/snapshots/"
+			dir('adapters') {
+				def filesToPush = findFiles(glob: '**.vsix')
+				sh "rsync -Pzrlt --rsh=ssh --protocol=28 ${filesToPush[0].path} ${UPLOAD_LOCATION}/snapshots/vscode-middleware-tools/"
+			}
+		}
+	}
+
+	if(publishToMarketPlace.equals('true')) {
+		timeout(time:5, unit:'DAYS') {
+			input message:'Approve deployment?', submitter: 'jrichter'
+		}
+
+		stage("Publish to Marketplace") {
+			dir('adapters') {
+				withCredentials([[$class: 'StringBinding', credentialsId: 'vscode_java_marketplace', variable: 'TOKEN']]) {
```

Comment:
@jrichter1 curious that `credentialsId: 'vscode_java_marketplace'`. Should we create one for adapters or use this.

Reply:
afaik these are the credentials for the Red Hat account in the marketplace, it's just that someone named them like this in jenkins

Reply:
I would rather not rename, it is a variable name from back in the days where we were not planning to extend our presence. All plugins use the same and a rename will cause 4-5 jobs that needs to be fixed for no apparent gain.